### ISSUE TITLE
Addition to Screening/Voice Mail?

### ISSUE THREAD

**Issue by whitedavidp** [thess/callattendant#6]
I have a friend whose telecom supplier offers a call screening service as part of his package (what a lucky guy!) and I have long coveted something his package does. When someone calls who is not recognized as either allow or block, it screens by saying something like "Your call is being screened. Prove you are a human and press # to be connected." If the user presses # then the phone rings through, otherwise they are disconnected...

I know that exactly this cannot be accomplished using Call Attendant. But I suspect that something close could be implemented. In the screening process, perhaps it can say something like "Your call is being screened. If you are a human, press # to be temporarily unblocked, hang up, and immediately call back." Then if # is pressed, Call Attendant can toggle the Permit Next Call state and then hang up? I feel like I have read somewhere about a commercial call-blocking box that does just this. So it is not without precedence.

When doing all this (perhaps it makes sense always when the Permit Next Call is toggled on) there should be a the means to automatically toggle Permit Next Call state back off within some timeout period? To me, it makes no sense to have Permit Next Call be turned on and simply left on indefinitely (at least I think that it what it does). So a configurable timeout of x minutes makes sense to have.

Looking at the code and considering my very limited python experience, I MIGHT be able to add this myself. From what I can tell, all of the "bones" for this are there. And the threading.Timer mechanism looks enough to me like Javascript's setTimeout() that it could be used easily enough. But at best, I would be stumbling around in the semi-darkness. So perhaps this would be a good enhancement?

Thanks

**Comment by roycewilliams** [comment]
Creating a voice response that requires a key press ("press 1") does appear to already be possible with the existing framework. You might take a closer look at the docs. Not where I can look it up right now, but will update this reply with specifics when I can.

**Comment by thess** [comment]
> If the user presses # then the phone rings through, otherwise they are disconnected...

You would need additional hardware to generate a local ring signal. Similar screening devices have their own ring/alert beeper and don't ring thru to your house phones.
>... I have read somewhere about a commercial call-blocking box that does just this

It's this: [Sentry Call Blocker](https://www.amazon.com/gp/product/B019ERMS90)

I had one for a couple of years but found it had an impossible user interface, a pitiful voice recorder and not very reliable. I had eventually decided to create my own. Along the way I discovered [emxsys/Callattendent](https://github.com/emxsys/Callattendant) which was a fantastic start for what I wanted - this repo is that WIP thanks to the huge boost from Bruce Shubert. 

"Permit next call" feature is just what it was intended for. Implementing a different mechanism with a specified timeout (or not) makes sense in the context you describe. You can always blacklist the number of the caller who can follow directions even if their intent is not what you think.

**Comment by whitedavidp** [comment]
Thanks for confirming all this.

Despite my reservations, I WAS going to try the Sentry v 3.x. But they simply are not available anywhere - even at the Sentry website. Plus, once I realized that the device relies on its own internal beeper and would NOT ring my actual house phones, I realized that it simply would NOT be a viable alternate anyhow.

Along my way, I also tried out entire phone sets from ATT and VTech only to find their screening mechanisms far from useful in that there is no way to batch enter numbers to their limited-size, allow lists. Even the ones that supposedly sync contact lists with a cell phone over BT do NOT include the resulting contacts in their screening for the land line. How stupid is that?

So I am very grateful to all who have worked so hard to make Call Attendant available and functional!
